src/realm/object-store/c_api/realm.hpp

```diff
@@ -69,47 +69,58 @@ class CBindingContext : public BindingContext {
 
 #if REALM_ENABLE_SYNC
 
-class CBindingThreadObserver : public realm::BindingCallbackThreadObserver {
+struct CBindingThreadObserver : public realm::BindingCallbackThreadObserver {
 public:
-    using ThreadCallback = util::UniqueFunction<void()>;
-    using ErrorCallback = util::UniqueFunction<void(const char*)>;
-
-    static CBindingThreadObserver& create()
+    CBindingThreadObserver(realm_on_object_store_thread_callback_t on_thread_create,
+                           realm_on_object_store_thread_callback_t on_thread_destroy,
+                           realm_on_object_store_error_callback_t on_error, realm_userdata_t userdata,
+                           realm_free_userdata_func_t free_userdata)
+        : m_create_callback_func{on_thread_create}
+        , m_destroy_callback_func{on_thread_destroy}
+        , m_error_callback_func{on_error}
+        , m_user_data{UserdataPtr(userdata, free_userdata)}
     {
-        static CBindingThreadObserver instance;
-        return instance;
     }
 
-    void set(ThreadCallback&& on_create, ThreadCallback&& on_destroy, ErrorCallback&& on_error)
+    virtual ~CBindingThreadObserver() = default;
+
+    // For testing: compare two CBindingThreadObserver instances to see if they have
+    // the same callback functions
+    bool operator==(const CBindingThreadObserver& other) const
```

Comment:
how is this used? Do we have multiple CBindingThreadObserver instances and need to check if they have the same callbacks?

Reply:
This is only for testing - it was an easy way to compare if two thread observer objects held the same callback functions when testing the `realm_sync_client_config_set_default_binding_thread_observer()` function. Since this function creates a new thread observer object and the callback functions are protected, the workaround was to create a duplicate object and then use that to compare the functions inside.

Reply:
I looked closer at the test. You basically want to make sure that the observer is the one intended. Why not auto-assign an id to each observer and check that value instead? (i.e, `observer.get_id() == expected_id`)

Reply:
I removed the second object and do direct comparisons on the values. I needed to add accessor functions to get the values inside the class, but that is not a big deal (and probably better than defining an `operator==` that may have unforeseen consequences).

Reply:
thanks